### ISSUE TITLE
When a sector is terminated while sealing on the miner side, deal remains in StorageDealSealing state for client.

### ISSUE THREAD

**Issue by dineshshenoy** [filecoin-project/lotus#4990]
From @dirkmc 

It seems that when a sector is terminated while sealing, deal remains in StorageDealSealing state.
For example a deal that went from StorageDealSealing -> StorageDealError (from the client's perspective) when the client restarted:

Nov 20 19:06:23  bafyreichan55tqeegojqjm3lnhjhtr6tqkd4l3t557tryg4kfbry4lpaqa  1367548  f02665   StorageDealSealing                Y (epoch 0)  Y (epoch 0)  baga6ea4seaqgowjicpl453l6matswfg2ftzjoxytcszrztmeaeg5z3ztfxyecoq  7.938 GiB  0.002082448 FIL   520612  false
Nov 20 19:06:23  bafyreichan55tqeegojqjm3lnhjhtr6tqkd4l3t557tryg4kfbry4lpaqa  1367548  f02665   StorageDealError                  Y (epoch 0)  Y (epoch 0)  baga6ea4seaqgowjicpl453l6matswfg2ftzjoxytcszrztmeaeg5z3ztfxyecoq  7.938 GiB  0.002082448 FIL   520612  false  error in deal activation: failed to set up called handler: called check error (h: 263798): failed to look up deal on chain: deal 1367548 not found

One idea that occurs to me is to periodically check if a deal has been terminated while it's in the StorageDealSealing state.

**Comment by hannahhoward** [comment]
Follow-up @dirkmc : We should probably timeout OnDealSectorPreCommitted and OnDealSectorCommitted on the StartEpoch for the deal proposal.

**Comment by hannahhoward** [comment]
^^ @magik6k do you think this will work? It seems to me if we miss the start epoch then the provecommit will fail either way?

**Comment by hannahhoward** [comment]
alternatively now that we are seperating out pre-commit, is there a fixed timeout between pre-commit and prove-commit?

**Comment by dirkmc** [comment]
Missing start epoch is one reason the deal will be terminated. Are there other reasons the deal might be terminated while it's in the AwaitingPreCommit / Sealing state?

**Comment by dirkmc** [comment]
Fixing this will most likely solve https://github.com/filecoin-project/lotus/issues/4977

**Comment by magik6k** [comment]
Yes, timeout on StartEpoch is the correct thing to do here, I believe this is when the deal will be removed from the market actor
